ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/deneb/block/BlockProcessorDenebTest.java

```diff
@@ -34,17 +39,43 @@ protected Spec createSpec() {
 
   @Test
   void shouldRejectCapellaBlock() {
-    final DataStructureUtil data = new DataStructureUtil(TestSpecFactory.createMinimalCapella());
     BeaconState preState = createBeaconState();
-    final SignedBeaconBlock block = data.randomSignedBeaconBlock(preState.getSlot().increment());
+    final SignedBeaconBlock block =
+        dataStructureUtil.randomSignedBeaconBlock(preState.getSlot().increment());
     assertThatThrownBy(
-            () ->
-                spec.processBlock(
-                    preState,
-                    block,
-                    BLSSignatureVerifier.SIMPLE,
-                    Optional.empty(),
-                    KzgCommitmentsProcessor.NOOP))
+            () -> spec.processBlock(preState, block, BLSSignatureVerifier.SIMPLE, Optional.empty()))
         .isInstanceOf(StateTransitionException.class);
   }
+
+  @Test
+  void shouldFailProcessingIfCommitmentsInBlockAreMoreThanMaxBlobsPerBlock() {
+    final BeaconState preState = createBeaconState();
+    final UInt64 slot = preState.getSlot().increment();
+    final int maxBlobsPerBlock = spec.getMaxBlobsPerBlock(slot).orElseThrow();
+    final BeaconBlockBody blockBody =
+        dataStructureUtil.randomBeaconBlockBodyWithCommitments(maxBlobsPerBlock + 1);
+    assertThatThrownBy(
+            () ->
+                spec.getBlockProcessor(slot)
+                    .validateExecutionPayload(preState, blockBody, Optional.empty()))
+        .isInstanceOf(BlockProcessingException.class)
+        .hasMessage("Number of kzg commitments in block exceeds max blobs per block");
+  }
+
+  @Test
+  @Override
+  public void shouldCreateNewPayloadRequest() throws BlockProcessingException {
+    final BeaconBlockBody blockBody = dataStructureUtil.randomBeaconBlockBodyWithCommitments(3);
+
+    final NewPayloadRequest newPayloadRequest =
+        spec.getBlockProcessor(UInt64.ONE).computeNewPayloadRequest(blockBody);
+
+    assertThat(newPayloadRequest.getExecutionPayload())
+        .isEqualTo(blockBody.getOptionalExecutionPayload().orElseThrow());
+    assertThat(newPayloadRequest.getVersionedHashes()).isPresent();
+    assertThat(newPayloadRequest.getVersionedHashes().get())
+        .hasSize(3)
+        .allMatch(
+            versionedHash -> versionedHash.isVersion(SpecConfigDeneb.VERSIONED_HASH_VERSION_KZG));
```

Comment:
Maybe also add a check that hashes match?

Reply:
done